.travis/rpmbuild_test

```diff
@@ -5,13 +5,13 @@
 OS_VERSION="$1"
 
 # build and install
-yum install -y libtool rpm-build make yum-utils squashfs-tools
+yum install -y libtool rpm-build make yum-utils
 ./autogen.sh
 yum-builddep -y singularity.spec
```

Comment:
This won't work... singularity.spec is not generated until after `./configure` runs.

If you specified `yum-builddep -y singularity.spec.in` would that grab the BuildRequires entries, or would it break because it isn't "complete"?

Reply:
Good point!
Any idea what's the magic trick why it still works in travis nevertheless?

Reply:
Don't know why that succeeds. There was a build earlier, where `yum-builddep` blew up because `singularity.spec` didn't exist... and it actually _shouldn't_ exist at that point...

Where is the singularity source being grabbed from? I don't see a checkout or tarball extraction prior to the `./autogen.sh` run.

Reply:
@jmstover see `.travis/centos_run` the source is bound into the docker container from `pwd` where it will be in the travis build environment:

```
docker run --privileged -d -ti -e "container=docker" -v /var/run/docker.sock:/docker.sock --network=host \
       -v /sys/fs/cgroup:/sys/fs/cgroup -v `pwd`:/build:rw  centos:${OS_VERSION}   /usr/sbin/init
```

Reply:
At least `rpmspec` doesn't like "incomplete" spec files. I'd think that something similar would happen in `yum-builddep`?

```
rpmspec -q --buildrequires --qf "[%{REQUIRENAME} ]" singularity.spec.in
error: line 28: Illegal char '@' (0x40) in: Version: @PACKAGE_VERSION@
error: query of specfile singularity.spec.in failed, can't parse
```

So going back to manual selection of the dependencies for the configure step?

Reply:
> So going back to manual selection of the dependencies for the configure step?

This has already gone wrong several times and will go wrong many times in the future, so I would strongly vote against it.

Reply:
Then for now one could just

```
sed 's/@PACKAGE_VERSION@/0/' < singularity.spec.in > singularity.spec.tmp
```

as the spec file is otherwise left alone by autotools.

Concerning the unclean build folder "problem", maybe move the `cp -a . /tmp/$DIST/singularity` from `centos_run` to `before_install`?